web/src/redux/reducers/alerts.js

```diff
@@ -0,0 +1,35 @@
+import { ALERT_RESOLVE, ALERT_SUCCESS } from '../actions/alert';
+
+const initialState = {
+  messages: []
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ALERT_SUCCESS:
+      return {
+        ...state,
+        messages: [
+          ...new Set(
+            state.messages.concat({
+              message: action.message,
+              variation: 'success'
+            })
+          )
+        ]
+      };
+    case ALERT_RESOLVE:
+      return removeMsg(action.message_index, state);
+    default:
+      return state;
+  }
+};
+
+export const removeMsg = (index, state) => {
```

Comment:
you don't have to, but you could simplify this function since you aren't using newState for anything else in the function to
```
export const removeMsg = (index, state) => state.messages.splice(index, 1);
```